packages/debug-test/MetroAutomation.ts

```diff
@@ -0,0 +1,58 @@
+
+// spawn: run process asynchronously (CreateProc?)
+// exec: spawns a shell and runs a command within that shell, passing the stdout and stderr to a callback function when complete.
+// Each of the methods returns a ChildProcess instance. 
+// fork
+// The child_process.fork() method is a special case of child_process.spawn() used specifically to spawn new Node.js processes. Like child_process.spawn()
+import { ChildProcess, exec, fork, spawn } from 'child_process';
```

Comment:
Metro and the RN CLI expose a native JavaScript APIs to run the Metro server. If they expose the capabilities needed, it may be more robust than piping out to a shell.. cc @tido64 @afoxman who would know more.

Reply:
Yeah, I considered this - and would much prefer it to spawning a separate Metro process. I currently lack the JS knowledge for the programming model that would allow Metro to operate "in the background" while my Jest test methods continue driving the host app and mock CDP debugger. If you have any tips/insights on how this would work, please share.

Reply:
Node has primitives for splitting the current JS context into a separate process via forking. https://nodejs.org/api/child_process.html#child_processforkmodulepath-args-options I think the overall process ownership model should be similar, but the managed process retains the JS context.

Reply:
Looking at th Metro server API, I think you might be able to start it and end it, in the setup/teardown of a test suite. https://facebook.github.io/metro/docs/getting-started/

Jest methods being async means that any await will defer to the event loop. So I think you may be able to run the Metro server in-process.